src/gamepad/gamepad-mappings-manager.vala

```diff
@@ -17,10 +19,19 @@ private class Games.GamepadMappingsManager : Object {
 		if (mappings == null)
 			mappings = new HashTable<string, string> (str_hash, str_equal);
 		try {
-			add_from_resource ("/org/gnome/Games/gamepads/gamecontrollerdb.txt");
+			add_from_resource (@"/org/gnome/Games/gamepads/$MAPPINGS_FILE_NAME");
 		}
 		catch (Error e) {
-			warning ("GamepadMappingsManager: Can't find gamecontrollerdb.txt: %s", e.message);
+			warning ("GamepadMappingsManager: Can't find %s: %s", MAPPINGS_FILE_NAME, e.message);
+		}
+		try {
+			var dir = Application.get_config_dir ();
+			var path = @"$dir/$MAPPINGS_FILE_NAME";
+			if (File.new_for_path (path).query_exists ())
```

Comment:
We avoid chaining calls in Games, especially creation calls. Instead we would create the file, then check if it exists. It's for a matter of readability. :)